coord/celestial.py

```diff
@@ -209,6 +209,82 @@ def from_xyz(x, y, z):
 
+    @staticmethod
+    def radec_to_xyz(ra, dec, r=1.):
+        """Convert ra, dec (in radians) to 3D x,y,z coordinates on the unit sphere.
+
+        The connection between (ra,dec) and (x,y,z) are given by the following formulae:
+        .. math::
+
+            x &= r \\cos(dec) \\cos(ra)  \\\\
+            y &= r \\cos(dec) \\sin(ra)  \\\\
+            z &= r \\sin(dec)
+
+        For a single ra,dec pair, the following are essentially equivalent:
+
+            >>> ra = 12*hours/radians       # May be any angle measured
+            >>> dec = 31*degrees/radians    # in radians
+
+            >>> CelestialCoord.radec_to_xyz(ra, dec)
+            (-0.85716730070211233, 1.0497271911386187e-16, 0.51503807491005416)
+
+            >>> CelestialCoord(ra * radians, dec * radians).get_xyz()
+            (-0.85716730070211233, 1.0497271911386187e-16, 0.51503807491005416)
+
+        However, the advantage of this function is that the input values may be numpy
+        arrays, in which case, the return values will also be numpy arrays.
+
+        :param ra:      The right ascension(s) in radians. May be a numpy array.
+        :param dec:     The declination(s) in radians. May be a numpy array.
+        :param r:       The distance(s) from Earth (default 1.). May be a numpy array.
+
+        :returns: x, y, z as a tuple.
+        """
+        import numpy
+        cosdec = numpy.cos(dec)
+        x = cosdec * numpy.cos(ra) * r
+        y = cosdec * numpy.sin(ra) * r
+        z = numpy.sin(dec) * r
+        return x,y,z
+
+    @staticmethod
+    def xyz_to_radec(x, y, z):
+        """Convert 3D x,y,z coordinates to ra, dec (in radians).
+
+        The connection between (ra,dec) and (x,y,z) are given by the following formulae:
+        .. math::
+
+            x &= r \\cos(dec) \\cos(ra)  \\\\
+            y &= r \\cos(dec) \\sin(ra)  \\\\
+            z &= r \\sin(dec)
+
+        For a single (x,y,z) position, the following are essentially equivalent:
+
+            >>> x = 0.839       # May be any any 3D location
+            >>> y = 0.123       # Not necessarily on unit sphere
+            >>> z = 0.530
+
+            >>> CelestialCoord.xyz_to_radec(x, y, z)
+            (0.14556615088111796, 0.55861619104852311)
+
+            >>> c = CelestialCoord.from_xyz(x, y, z)
+            >>> c.ra.rad, c.dec.rad
+            (0.145566150881118, 0.558616191048523)
+
+        However, the advantage of this function is that the input values may be numpy
+        arrays, in which case, the return values will also be numpy arrays.
+
+        :param x:       The x position(s) in 3 dimensions. May be a numpy array.
+        :param y:       The y position(s) in 3 dimensions. May be a numpy array.
+        :param z:       The z position(s) in 3 dimensions. May be a numpy array.
+
+        :returns: ra, dec as a tuple.
+        """
+        import numpy
+        ra = numpy.arctan2(y, x)
+        dec = numpy.arctan2(z, numpy.sqrt(x**2+y**2))
+        return ra,dec
```

Comment:
It's about 10% more expensive here to also calculate `r=np.sqrt(x**2+y**2+z**2)`.  Maybe make that at least an optional output for symmetry?